src/compile/scale.ts

```diff
@@ -43,8 +47,43 @@ export function compileScales(channels: Channel[], model: Model) {
         }
       });
 
-      return scaleDef;
-    });
+      // add additional scales needed to support ordinal color scales also for legends
+      if (channel === COLOR && fieldDef.legend && (fieldDef.type === ORDINAL || fieldDef.bin || fieldDef.timeUnit)) {
+        scales.push({
+          name: INVERSE_RANK,
+          type: ORDINAL,
+          domain: {
+            data: model.dataTable(),
+            field: model.field(COLOR, (fieldDef.bin || fieldDef.timeUnit) ? {} : {prefn: 'rank_'}), sort: true
+          },
+          range: {data: model.dataTable(), field: model.field(COLOR), sort: true}
+        });
+
+        // bin needs an additional scale for labels because we want labels in legends
```

Comment:
Hmm instead of because .... because ... 

can you have one "because": 

"because we need to map bin_start to bin_range in legend's "labels